Env: Fixed `run` Argument Passing

ObliviousHarmony opened the pull request:
## What?

As it stands, you need to pass options to `wp-env run` with the entire string having been quoted. This is because of `yargs/yargs` consuming any options even if they aren't defined. This causes problems when trying to encapsulate scripts in `package.json` files since you can't easily inject more options into the quoted string in the script.

Closes #32929.

## Why?

This will make it easier to write `wp-env run` commands in `package.json` scripts that can accept more options. Something like `wp-env run tests-wordpress ls -la` will now work without being quoted. With this pull request, you can now pass options.

## How?

This works by enabling the `unknown-options-as-args` parser option. This results in any unknown options that are given being treated as a positional argument. In the case of a variadic argument like `[command...]`, this means we can write commands conveniently.

As a note, `--help` and `--version` cannot be passed since they are known options. This will also result in other commands failing to execute if unknown flags are passed inappropriately in place of positional arguments.

## Testing Instructions
1. Run `npm run wp-env -- run ls -la`
2. With this pull request the command will execute as expected. Without it, there will be no `-la` option.3.

ObliviousHarmony committed:
Fixed `run` Argument Passing

By enabling the `unknown-options-as-args` parser
option, any options passed to the command that
are not registered will be given to the command
as arguments. This lets us pass them to the
container without quoting them.

Note: --help and --version can't be passed since they
aren't unknown. This is probably fine though and we
can handle that using quotes when necessary.

noahtallen reviewed (approved):
Love it, very simple way to solve the problem! And quotes still work if the argument matches a wp-env argument as a fallback.

I think we just need to update the README file. For example, we have this notice about needing quotation marks in some cases: https://github.com/WordPress/gutenberg/blob/c088c448b52674b121e18b3eff3fb250547cdbbd/packages/env/README.md#L302-L312, so we can probably remove that and note that quotation marks can be used as a fallback if wp-env consumes an arg instead of passing it in.

Edit: oh and we should also note it as an enhancement in the changelog :)

ObliviousHarmony committed:
Documented Quotes Behavior

Since we no longer require the quotes, this
commit updates the documentation to
explain what cases they are required to
be used. This will keep it from being a
surprise if a command does not behave as
expected.

ObliviousHarmony committed:
Added Changelog Entry

ObliviousHarmony commented:
Thanks, @noahtallen,

I've added a changelog and updated the documentation.
